src/box/session.cc

```diff
@@ -164,6 +166,35 @@ session_create(enum session_type type)
 	return session;
 }
 
+struct session *
+next_session(struct session *session)
+{
+	struct rlist *next_rlist;
+	if (session == NULL)
+		next_rlist = rlist_first(&active_sessions);
+	else
+		next_rlist = rlist_next(&session->in_active_list);
+	if (next_rlist == &active_sessions)
+		return NULL;
+	else
+		return container_of(next_rlist, struct session, in_active_list);
+}
+
+bool
+is_shutdown_ready(struct session *session)
+{
+	bool is_wait_client_shutdown = !session->graceful_shutdown ||
+		(session->graceful_shutdown && session->client_shutdown_got);
```

Comment:
You don't need to check `session->graceful_shutdown` two times.
bool is_wait_client_shutdown = !session->graceful_shutdown || session->client_shutdown_got;

Reply:
Removed extra `session->graceful_shutdown` and added comments in the function declaration.